src/OAuthClient/HoldRequestClient.php

```diff
@@ -59,56 +63,88 @@ public static function validateProcessed($processed)
     public static function validateSuccess($success)
     {
         if (!isset($success)) {
-            throw new APIException(
-                'Success flag not set',
-                'Success flag is not set.',
-                0,
+            throw new NonRetryableException(
+                'Not Acceptable: Success flag not set',
+                'Not Acceptable: Success flag is not set.',
+                406,
                 null,
-                500,
-                new ErrorResponse(500, 'success-flag-not-set', 'Success flag is not set.')
+                406,
+                new ErrorResponse(406, 'success-flag-not-set', '406 Not Acceptable: Success flag is not set.')
             );
         }
 
         return true;
     }
 
     /**
-     * @param string $holdRequestId
-     * @return HoldRequest
-     * @throws APIException
+     * @param $holdRequestId
+     * @return null|HoldRequest
+     * @throws NonRetryableException
+     * @throws RetryableException
      */
-    public static function getHoldRequestById($holdRequestId = '')
+    public static function getHoldRequestById($holdRequestId)
     {
         self::validateRequestId($holdRequestId);
 
         $url = Config::get('API_HOLD_REQUEST_URL') . '/' . $holdRequestId;
 
-        APILogger::addDebug('Retrieving hold request by id',  (array) $url);
+        APILogger::addDebug('Retrieving hold request by id', (array) $url);
 
-        $response = self::get($url);
+        try {
+            $response = self::get($url);
 
-        $response = json_decode((string)$response->getBody(), true);
+            $statusCode = $response->getStatusCode();
 
-        APILogger::addDebug('Retrieved hold request by id', $response['data']);
+            $response = json_decode((string)$response->getBody(), true);
 
-        if ($response['statusCode'] !== 200) {
-            APILogger::addError(
-                'Failed',
-                array('Failed to retrieve Hold Request ', $holdRequestId, $response['type'], $response['message'])
+            APILogger::addDebug('Retrieved hold request by id', $response['data']);
+
+            // Check statusCode range
+            if ($statusCode === 200) {
+                return new HoldRequest($response['data']);
+            } else {
+                APILogger::addError(
+                    'Failed',
+                    array('Failed to retrieve Hold Request ', $holdRequestId, $response['type'], $response['message'])
+                );
+                return null;
+            }
+        } catch (ServerException $exception) {
+            throw new RetryableException(
+                'Server Error from ' . __FUNCTION__,
+                'Server Error from ' . __FUNCTION__,
```

Comment:
Same as above for exception params; second param can be null instead of doubling the message.